builtin/mainmenu/tab_online.lua

```diff
@@ -264,19 +265,52 @@ end
 
 --------------------------------------------------------------------------------
 
+-- Separates by space characters and handles special prefixes
+-- (words with special prefixes need an exact match and none of them can contain spaces
+-- or quotation marks)
+local function parse_search_word(query, text)
+	for word in text:gmatch("%S+") do
+		if string.sub(word, 0, 4) == "mod:" then
+			table.insert(query.mods, string.sub(word, 5))
+		elseif string.sub(word, 0, 7) == "player:" then
+			table.insert(query.players, string.sub(word, 8))
+		elseif string.sub(word, 0, 5) == "game:" then
+			query.game = string.sub(word, 6)
+		else
+			table.insert(query.keywords, word:lower())
+		end
+	end
+end
+
+local function parse_search_input(input)
+	local query = {keywords = {}, mods = {}, players = {}}
+
+	-- Separate quote enclosed parts
```

Comment:
Personal taste: I would find this easier to read if you

1. Remove (and process) quoted parts via `gsub` and a function [^1]
2. Then pass the rest to `parse_search_word`.

[^1]: Or, if you find it cleaner, process them via `gmatch`, remove them via `gsub`.

Reply:
It now uses gsub for the quoted parts. (Not sure if it makes a performance different, but it shouldn't matter anyway.)